src/controllers/task.controller.ts

```diff
@@ -13,11 +13,70 @@ export const getTaskByStatus = async (req: Request, res: Response): Promise<void
 
   const status = req.params?.status;
 
-  if (status !== ETaskStatus.Complete && status !== ETaskStatus.Incomplete) {
+  if (![ETaskStatus.Complete,ETaskStatus.Incomplete].some(t=>t === status)) {
     res.status(400).send('Not a valid status');
   } else {
     const tasks = await taskService.getTaskByStatus(status);
     res.status(200).send(tasks);
   }
 
-}
\ No newline at end of file
+}
+
+export const addTask = async (req: Request, res: Response): Promise<void> => {
+    
+  const data: string = req.body?.task;
+
+  if (!data){
+    res.status(400).send('Missing required field: task');
+    return;
+  } else {
+    const newTaskID = await taskService.addTask(data);
+    res.status(200).send({ id: newTaskID.id });
+  }
+
+}
+
+export const deleteTaskByID = async (req: Request, res: Response): Promise<void> => {
+    
+  const id = req.params?.id;
+  try{
+    await taskService.deleteTaskByID(id);
+    res.status(200).send();
+  }catch(err){
+    res.status(400).send(err);
```

Comment:
Here either HTTP 404 or HTTP 500. HTTP 400 is bad request which is not for this scenario

Reply:
Also these type of error can be easily handled using an error middleware. 
Check out this one https://expressjs.com/en/guide/error-handling.html
So instead of handle the error manually, you can just call `next(err)` and let the middleware handle it for you